src/Service/Resolvers/MsSqlQueryExecutor.cs

```diff
@@ -194,5 +197,59 @@ public override string GetSessionParamsQuery(HttpContext? httpContext, IDictiona
 
             return sessionMapQuery.ToString();
         }
+
+        /// <inheritdoc/>
+        public override async Task<DbResultSet> GetMultipleResultSetsIfAnyAsync(
+            DbDataReader dbDataReader, List<string>? args = null)
+        {
+            // From the first result set, we get the count(0/1) of records with given PK.
+            DbResultSet resultSetWithCountOfRowsWithGivenPk = await ExtractResultSetFromDbDataReader(dbDataReader);
+            int numOfRecordsWithGivenPK = (int)resultSetWithCountOfRowsWithGivenPk.Rows.FirstOrDefault()!.Columns[MsSqlQueryBuilder.COUNT_ROWS_WITH_GIVEN_PK]!;
+
+            // The second result set holds the records returned as a result of the executed update/insert operation.
+            DbResultSet? dbResultSet = await dbDataReader.NextResultAsync() ? await ExtractResultSetFromDbDataReader(dbDataReader) : null;
+
+            if (dbResultSet is null)
+            {
+                // This can only happen in the case where we are dealing with table/view with auto-gen PK.
+                // In this case, insert cannot happen.
+                // Also since dbResultSet is null, update couldn't execute as well because there was no record corresponding to given PK.
+                string prettyPrintPk = args![0];
```

Comment:
>string prettyPrintPk = args![0] [](http://example.com/codeflow?start=16&length=31)

can you add comment for how args can be guaranteed to be not null when dbResultSet is null?

Reply:
The call to this method is made only from here:

https://github.com/Azure/data-api-builder/blob/c69025cf8cd39df3cf1dee06565f9c18641ba62c/src/Service/Resolvers/SqlMutationEngine.cs#L670

where args is explicitly initialized with a list of size 2. In the signature, args is nullable only so that the different delegates conform in the signature.

Reply:
What if someone uses the same function in future and does not initialize `args`? What stops them from not doing so? We should add a check for `args` is not null